apps/meteor/client/views/admin/manageSubscription/components/UpgradeButton.tsx

```diff
@@ -0,0 +1,36 @@
+import { Button, ButtonGroup, Throbber } from '@rocket.chat/fuselage';
+import type { ButtonProps } from '@rocket.chat/fuselage/dist/components/Button/Button';
+import type { ReactElement } from 'react';
+import React, { memo } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { useExternalLink } from '../../../../hooks/useExternalLink';
+import { useCheckoutUrl } from '../hooks/useCheckoutUrl';
+
+type UpgradeButtonProps = {
+	i18nKey?: string;
+} & Partial<ButtonProps>;
+
+const UpgradeButton = ({ i18nKey = 'Upgrade', ...props }: UpgradeButtonProps): ReactElement => {
+	const { t } = useTranslation();
+	const handleExternalLink = useExternalLink();
+	const { isLoading, url } = useCheckoutUrl();
+
+	const handleBtnClick = () => {
+		if (isLoading) {
+			return;
+		}
+
+		handleExternalLink(url);
+	};
+
+	return (
```

Comment:
Instead of `<Throbber /`> why not just disable the `Button` and change the text?

Reply:
Because it's better for loading state

Reply:
better for who? IMO it's clean and elegant

Reply:
UI and I prefer also... but they will work on a loading button to improve in that scenario